prov/sockets/src/sock_progress.c

```diff
@@ -2089,6 +2098,20 @@ static int sock_pe_progress_tx_entry(struct sock_pe *pe,
 	int ret;
 	struct sock_conn *conn = pe_entry->conn;
 
+	if (sock_comm_is_disconnected(pe_entry)) {
```

Comment:
Thanks for the review,
I have added a new patch on top of the patch series. As you mentioned before, this patch marks the socket as disconnected as soon as send() returns EPIPE. 
@shefty, do you agree with this change?